client/my-sites/stats/stats-module/all-time-nav.jsx

```diff
@@ -69,66 +72,82 @@ export const StatsModuleSummaryLinks = ( props ) => {
 		recordStats( item );
 	};
 
-	// Template for standard range options (7, 30, Quarter, Year, All Time).
-	const summaryPath = `/stats/day/${ path }/${ siteSlug }?startDate=${ moment().format(
-		'YYYY-MM-DD'
-	) }&summarize=1&num=`;
+	const getSummaryPath = useMemo( () => {
+		return ( numberDays ) => {
+			let url = window.location.href;
+			let queryParams = new URLSearchParams( url.split( '?' )[ 1 ] || '' );
 
-	// Path for summary or custom range option. ie: The first button in the row.
-	// Defaults to one day/week/month/year.
-	let summaryPeriodPath = `/stats/${
-		period.period
-	}/${ path }/${ siteSlug }?startDate=${ period.endOf.format( 'YYYY-MM-DD' ) }`;
-	// Override if custom range was used in query.
-	if ( query.start_date ) {
-		summaryPeriodPath = `/stats/${ period.period }/${ path }/${ siteSlug }?startDate=${ query.start_date }&endDate=${ query.date }`;
-	}
+			if ( isOdysseyStats ) {
```

Comment:
A unified way to deal with Odyssey Stats parameters is to read params from `context`.